internal/app/agent/service.go

```diff
@@ -168,41 +177,76 @@ func (a *agent) Control(uuid, cmdStr string) error {
 		return err
 	}
 
-	payload, err := encodeSenML(uuid, cmd, resp)
-	if err != nil {
-		return err
-	}
-
-	if err := a.Publish(a.config.Agent.Channels.Control, string(payload)); err != nil {
-		return err
-	}
-
-	return nil
+	return a.processResponse(uuid, cmd, resp)
 }
 
 // Message for this command
-// "[{"bn":"1:", "n":"config", "vs":"export, /configs/export/config.toml, config_file_content"}]"
+// [{"bn":"1:", "n":"services", "vs":"view"}]
+// [{"bn":"1:", "n":"config", "vs":"save, export, filename, filecontent"}]
 // config_file_content is base64 encoded marshaled structure representing service conf
 // Example of creation:
 // 	b, _ := toml.Marshal(cfg)
 // 	config_file_content := base64.StdEncoding.EncodeToString(b)
 func (a *agent) ServiceConfig(uuid, cmdStr string) error {
 	cmdArgs := strings.Split(strings.Replace(cmdStr, " ", "", -1), ",")
-	if len(cmdArgs) < 3 {
+	if len(cmdArgs) < 1 {
 		return errInvalidCommand
 	}
+	resp := ""
+	cmd := cmdArgs[0]
+
+	switch cmd {
+	case view:
+		services, err := json.Marshal(a.Services())
+		if err != nil {
+			return err
+		}
+		resp = string(services)
+	case save:
+		if len(cmdArgs) < 4 {
+			return errInvalidCommand
+		}
+		service := cmdArgs[1]
+		fileName := cmdArgs[2]
+		fileCont := cmdArgs[3]
+		if err := a.saveConfig(service, fileName, fileCont); err != nil {
+			return err
+		}
+	}
+	return a.processResponse(uuid, cmd, resp)
+}
 
-	service := cmdArgs[0]
-	fileName := cmdArgs[1]
-	fileCont, err := base64.StdEncoding.DecodeString(cmdArgs[2])
+func (a *agent) processResponse(uuid, cmd, resp string) error {
+	payload, err := encodeSenML(uuid, cmd, resp)
 	if err != nil {
 		return err
 	}
-	c := &export.Config{}
+	if err := a.Publish(a.config.Agent.Channels.Control, string(payload)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (a *agent) saveConfig(service, fileName, fileCont string) error {
+	switch service {
+	case "export":
```

Comment:
Consider moving service names in const's, maybe use iota.

Reply:
i think that we will support additional services later, so when I get there will think how to redefine this

Reply:
I would prefer that we do it now. You van create new enum type (with `iota`), you have an exaple in `mproxy`.

It will also be useful as a self-documentig code, specifying which commands are available.

Reply:
Here is the example: https://github.com/mainflux/mproxy/blob/master/pkg/mqtt/session.go#L16

Even without `iota`, `type command string` would help, and all command can be listed under one const.

BTW, I can see you are using them like consts in `internal/pkg/conn/conn.go` - if these are used in several places, then create a shared package to hold these consts and import them in different packages,

Reply:
I would do the same as he did in `ServiceConfig`, i.e create const for this values.

Reply:
I cannot use iota, @drasko we talked about this
```
// Message for this command
// [{"bn":"1:", "n":"services", "vs":"view"}]
// [{"bn":"1:", "n":"config", "vs":"save, export, filename, filecontent"}]
// config_file_content is base64 encoded marshaled structure representing service conf
// Example of creation:```

export comes from message and service refers to export service,like agent is saving configuration for export service.

as for type command string I had that https://github.com/mainflux/agent/blob/68c29b1012aa7d0992c6e109fe7795fa10b17383/internal/pkg/mqtt/sub.go#L20

but removed as part of resolving comments to previous PR